### PR TITLE
Include Request Body & All Http Methods

### PR DESCRIPTION
**Pull request by mblydenburgh**
- update gui to have a text window for inserting request body.
- add deafult content-type header.
- bring in api changes from tqhoughton branch for suppporting other methods.
- start generating request ids on submission for use in db.

### CODE REVIEW

File: gui/src/main.rs [hunk 1]
```diff
@@ -137,10 +139,19 @@ impl App for Gui {
                 ui.label("URL:");
                 ui.text_edit_singleline(&mut self.config.url);
                 if ui.button("Submit").clicked() {
+                    let body = if self.config.selected_http_method != HttpMethod::GET {
+                        Some(serde_json::from_str(&self.config.body_str).expect("Body is invalid json"))
```
Comment (mblydenburgh) [inline gui/src/main.rs:143]: Need to still gracefully handle this invalid parsing, if it fails it panics. the .expect() just leaves a nicer message in the terminal.

File: gui/src/main.rs [hunk 2]
```diff
@@ -37,7 +38,8 @@ impl Default for GuiConfig {
             active_window: ActiveWindow::REQUEST,
             request_window_mode: RequestWindowMode::BODY,
             selected_http_method: HttpMethod::GET,
-            url: String::from("https://localhost:3000"),
+            url: String::from("https://httpbin.org/json"),
```
Comment (mblydenburgh) [inline gui/src/main.rs:41]: This httpbin site I came across is pretty sweet for testing this! there are routes for each method like /get, /post, etc.

File: gui/src/main.rs [hunk 3]
```diff
@@ -152,21 +163,49 @@ impl App for Gui {
...
         });
         match self.config.request_window_mode {
             RequestWindowMode::BODY => {
+                TopBottomPanel::top("request_panel")
+                    .resizable(true)
+                    .min_height(250.0)
+                    .show(ctx, |ui| {
+                        ScrollArea::vertical().show(ui, |ui| {
+                            ui.add(
+                                TextEdit::multiline(&mut self.config.body_str)
```
Comment (mblydenburgh) [inline gui/src/main.rs:185]: Pretty nice that we get a formatted code block that you can hit enter in and not lose focus. Only thing left would be to figure out a how to implement a code formatting button to beautify the json.

File: gui/src/main.rs [hunk 4]
```diff
@@ -152,21 +163,49 @@ impl App for Gui {
...
+                    self.config.request_window_mode = RequestWindowMode::HEADERS;
+                }
+                if ui.button("Body").clicked() {
+                    self.config.request_window_mode = RequestWindowMode::BODY;
+                }
             });
         });
         match self.config.request_window_mode {
             RequestWindowMode::BODY => {
+                TopBottomPanel::top("request_panel")
```
Comment (mblydenburgh) [inline gui/src/main.rs:179]: Currently this panel seems to be half width? Would be ideal if we could stretch this across the entire width.

File: gui/src/main.rs (hunk 3 above)
Comment (tqhoughton) [reply gui/src/main.rs:185]: I think I remember the docs page mentioning how to install language packs?

**Review by tqhoughton — APPROVED** [review]

**Pull request merged**